src/Period.php

```diff
@@ -887,22 +887,37 @@ public function merge(Period $arg)
      * </ul>
      *
      * @param DateInterval|int|string $interval The interval
+     * @param bool                    $backwards Optionnal - iterates counterwise
      *
      * @return Generator|Period[]
      */
-    public function split($interval)
+    public function split($interval, $backwards = false)
     {
-        $startDate = $this->startDate;
-        $interval = static::filterDateInterval($interval);
-        do {
-            $endDate = $startDate->add($interval);
-            if ($endDate > $this->endDate) {
-                $endDate = $this->endDate;
-            }
-            yield new static($startDate, $endDate);
-
-            $startDate = $endDate;
-        } while ($startDate < $this->endDate);
+        if (false === $backwards) {
+            $startDate = $this->startDate;
+            $interval = static::filterDateInterval($interval);
+            do {
+                $endDate = $startDate->add($interval);
+                if ($endDate > $this->endDate) {
+                    $endDate = $this->endDate;
+                }
+                yield new static($startDate, $endDate);
+
+                $startDate = $endDate;
+            } while ($startDate < $this->endDate);
+        } else {
+            $endDate = $this->endDate;
+            $interval = static::filterDateInterval($interval);
+            do {
```

Comment:
This block is repeated in both conditionals, but with changed variables - can simply move to a private method